tests/conftest.py

```diff
@@ -244,8 +311,16 @@ def test_session_tmp_path(test_fc_session_root_path):
 @pytest.fixture
 def results_file_dumper(request):
     """Yield the custom --dump-results-to-file test flag."""
+    generate_baselines = False
+    dump_results = False
+
     if request.config.getoption("--dump-results-to-file"):
-        return ResultsFileDumper(request.node.originalname)
+        dump_results = True
+    if request.config.getoption("--generate-baselines"):
+        generate_baselines = True
+
+    if dump_results or generate_baselines:
+        return ResultsFileDumper(request, dump_results, generate_baselines)
```

Comment:
This method could always return a `ResultsFileDumper` instance with a dummy functionality for `dump` whenever dumping is not enabled.

Reply:
@alindima actually, I've given this a second thought and it's not really needed to add the checks anywhere in the tests.

Hooks can be added when tests start/end, like in the JSON test report thing, and the dumping can be done implicitly.